### PR TITLE
[runtime] Fix crash reporting configure options

### PR DESCRIPTION
**Pull request by alexanderkyte**

### CODE REVIEW

**Comment by alexanderkyte** [comment]
@monojenkins build failed

File: configure.ac [hunk 1]
```diff
@@ -1000,15 +1000,19 @@ fi
...
 
-AC_ARG_WITH(crash-privacy,       [  --with-crash_privacy=yes,no         If you want your crashes to not include names of symbols present in the binary. ], [], [with_crash_privacy=no])
+AC_ARG_WITH(crash-privacy,       [  --with-crash_privacy=yes,no         If you want your crashes to not include names of symbols present in the binary. ], [], [with_crash_privacy=yes])
 AC_ARG_WITH(structured-crashes,  [  --with-structured_crashes=yes,no    If you want your unmanaged crashes to result in a small crash dump file. ],        [], [with_structured_crashes=yes])
 
-if test "x$with_crash_privacy" = "xno"; then
+if test "x$with_crash_privacy" = "xyes"; then
 AC_DEFINE(MONO_PRIVATE_CRASHES,1,[Do not include names of unmanaged functions in the crash dump])
+CFLAGS="$CFLAGS -DMONO_PRIVATE_CRASHES"
+CXXCFLAGS="$CXXCFLAGS -DMONO_PRIVATE_CRASHES"
```
Comment (lewurm) [inline configure.ac]: why is this needed when there is `AC_DEFINE` for it?

**Comment by alexanderkyte** [comment]
@monojenkins build Failed

**Comment by alexanderkyte** [comment]
bug-323114.exe timing out on CI

**Comment by alexanderkyte** [comment]
@monojenkins build Failed

**Comment by alexanderkyte** [comment]
@monojenkins build Failed
